From 7fa5855b3cd44df677af27d155d73dffd9e443fd Mon Sep 17 00:00:00 2001
From: Kai Sommerfeld <kai.sommerfeld@gmx.com>
Date: Sun, 27 Oct 2019 19:48:16 +0100
Subject: [PATCH] [profiles] Do always stop PVR services on logoff. Restart PVR
 services only when not loading the master profile for the login screen.

---
 xbmc/profiles/ProfileManager.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/xbmc/profiles/ProfileManager.cpp b/xbmc/profiles/ProfileManager.cpp
index 8161e1ec1424..2d23bcc3eecb 100644
--- a/xbmc/profiles/ProfileManager.cpp
+++ b/xbmc/profiles/ProfileManager.cpp
@@ -404,8 +404,9 @@ void CProfileManager::FinalizeLoadProfile()
   // Restart context menu manager
   contextMenuManager.Init();
 
-  // restart PVR services
-  pvrManager.Init();
+  // Restart PVR services if we are not just loading the master profile for the login screen
+  if (m_profileLoadedForLogin || m_currentProfile != 0 || m_lastUsedProfile == 0)
+    pvrManager.Init();
 
   favouritesManager.ReInit(GetProfileUserDataFolder());
 
@@ -443,6 +444,9 @@ void CProfileManager::LogOff()
   if (CVideoLibraryQueue::GetInstance().IsRunning())
     CVideoLibraryQueue::GetInstance().CancelAllJobs();
 
+  // Stop PVR services
+  CServiceBroker::GetPVRManager().Stop();
+
   networkManager.NetworkMessage(CNetwork::SERVICES_DOWN, 1);
 
   LoadMasterProfileForLogin();
